lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -1388,8 +1388,22 @@ private FoundHits queryHits(
                                                                       sr.ascending ? "asc" : "desc"));
             sort.add(sortOrderWrapper);
         }
-        String indexName = objectType == null ? getAnyIndexPattern() : checkIndex(objectType);
-        String urlPath = "/" + indexName + "/" + getDataTableName() + "/_search";
+
+        StringBuffer indexName = new StringBuffer();
+        for(int ii=0; ii<objectTypes.size(); ii++) {
+
+            String objectType = objectTypes.get(ii);
+
+            indexName.append(
+                    objectType == null ? getAnyIndexPattern() : checkIndex(objectType)
+            );
+
+            if( ii < objectTypes.size()-1 )
```

Comment:
Simpler implementation would be something like

String indexName = String.join(",", objectTypes.stream().map(s -> checkIndex(s)).collect(Collectors.toList()));

Reply:
I tried this but it got all convoluted because checkIndex throws an exception and the syntax for handling that exception looked real messy. 

I am not that adept with lamba expressions in Java yet. Perhaps I should spend one of the weekly 10% time on this.

Reply:
Most of the benefit comes from the String.join() IMO. Lambdas and checked exceptions don't play well so try:

final List<String> r = new LinkedList<>();
for (final String type: objetTypes) {
    r.add(checkIndex(type));
}
final String indexName = String.join(",", r);

Reply:
Done.